http.go

```diff
@@ -120,7 +141,7 @@ func (req *APIRequest) Do() (*http.Response, error) {
 		return res, nil
 	}
 
-	return nil, NewRequestFailedError(res, req)
+	return res, NewRequestFailedError(res, req)
```

Comment:
Do we really want to return `res` in case of error? Doesn't it increase the risk of someone forgetting to check the returned error and use the returned result as if it were successful?